### ISSUE TITLE
clang-tidy misreports indentation with tabs and spaces

### ISSUE THREAD

**Issue by orangebowlerhat** [llvm/llvm-project#87202]
clang-tidy sometimes reports the non-matching indent on if-else where the indentation is correct.

I'm using tabs for indents, 4 characters wide. The `if` and `else` have the same number of tabs. The problem occurs when the `else` has a stray space after one of the tabs. This does not change the indentation at all, but clang-tidy reports it as a mismatch.

This code -
```
int foo(bool b) {
	if (b)
		return 42;
 	else
		return 0;
}

```
Produces this warning -
```
$ clang-tidy --checks=readability-misleading-indentation test.cpp
1 warning generated.
test.cpp:4:3: warning: different indentation for 'if' and corresponding 'else' [readability-misleading-indentation]
    4 |         else
      |         ^
```

AFAIK there is no case in which the space could cause the indentation not to match. Tabs have to be at least one character wide. So while the indentation is not the same characters, it does not mismatch or mislead.

See also https://github.com/clangd/clangd/issues/1987

Forgive me if this is already resolved, I did look but couldn't find anything which looked the same.

**Comment by vbvictor** [comment]
https://godbolt.org/z/nT4je3o55
